src/Queue/Connectors/RabbitMQConnector.php

```diff
@@ -106,7 +108,12 @@ protected function createQueue(
             case 'horizon':
                 return new HorizonRabbitMQQueue($connection, $queue, $dispatchAfterCommit, $options);
             default:
-                return new $worker($connection, $queue, $options);
+                switch ($customWorkerType) {
+                    case 'with_dispatch_after_commit':
+                        return new $worker($connection, $queue, $dispatchAfterCommit, $options);
+                    default:
+                        return new $worker($connection, $queue, $options);
+                }
```

Comment:
this feels like a hack rather than a solution to the problem.

would like to see a move towards using factory make functions that return a class so we don't have to rely on params being perfectly matched up as long as they all share an underlying contract. this is something @khepin and I have discussed but are yet to really dig into a good solution yet.

Reply:
well, the library right now have a major bug which blocks real use of it (because call incompatible with Laravel )
I would prefer it fixed in some compatible way (right now we rely on custom fork)

Reply:
nope, nope, nope

This does not only feels like a hack... it is a hack.

Reply:
> the library right now have a major bug which blocks real use of it

Nothing is preventing you from creating you own `$worker` class compatible with the underlying Queue contracts and in the constructor doing what you need to. You do not **have** to extend `VladimirYuldashev\LaravelQueueRabbitMQ\Queue\RabbitMQQueue` to use this library.

I agree its not great that the constructors are no aligned but the fix is not a hack to make it work how you want it to but update to use factories to work with everyones code.